pkg/connector/repository.go

```diff
@@ -202,6 +208,10 @@ func (o *repositoryResourceType) Grants(
 		}
 		teams, resp, err := o.client.Repositories.ListTeams(ctx, orgName, resource.DisplayName, opts)
 		if err != nil {
+			if resp.StatusCode == http.StatusForbidden {
+				l.Warn("insufficient access to list teams", zap.String("repository", resource.DisplayName))
+				return nil, "", nil, nil
```

Comment:
we shouldn't return empty string for the next page token, if someone has permissions to list repository collaborators but not list repository teams, we are going to skip those grants. instead you can  do

```
err = bag.Next("")
if err != nil {
	return nil, "", nil, err
}
pageToken, err := bag.Marshal()
if err != nil {
	return nil, "", nil, err
}
return nil, pageToken, nil, nil